pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/SplitSegmentCommitter.java

```diff
@@ -70,4 +68,15 @@ public SplitSegmentCommitter(Logger segmentLogger, ServerSegmentCompletionProtoc
     }
     return commitEndResponse;
   }
+
+  // Return false iff the segment upload fails.
+  protected boolean uploadSegment(File segmentTarFile, SegmentUploader segmentUploader,
+      SegmentCompletionProtocol.Request.Params params) {
+    URI segmentLocation = segmentUploader.uploadSegment(segmentTarFile, new LLCSegmentName(params.getSegmentName()));
+    if (segmentLocation != null) {
+      params.withSegmentLocation(segmentLocation.toString());
```

Comment:
Why are we changing request parameters while indicating response? Can we remove this?

Reply:
done together with change with the method return type.

Reply:
Done.